contents/security/passwordStorage.md

```diff
@@ -0,0 +1,118 @@
+<frontmatter>
+  title: Introduction to Password Storage
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+{{ navbar | safe }}
+
+<div class="website-content">
+
+# Introduction to Password Storage
+
+Author: [Jeremy Choo](https://github.com/ChooJeremy) <br>
+Reviewers: [Amrut Prabhu](https://github.com/amrut-prabhu), [Marvin Chin](https://github.com/marvinchin), [Tan Zhen Yong](https://github.com/Xenonym), [Wang Junming](https://github.com/junming403)
+
+## Overview
+
+While developing your own application, it is very common to implement login systems to be able to identify users. Whenever there is a need to store user-specific information, developers typically implement user accounts to identify these users separately. You might have seen this mechanism implemented on other websites - using registration forms and requring you to login with a username and password to identify you for the <trigger for="pop:session"><a href="https://support.google.com/analytics/answer/2731565?hl=en">session</a></trigger>.
+
+<popover id="pop:session" title="A _session_  is used to broadly describe a user's visit" placement="top">
+  <div slot="content">
+Generally, a session starts when a user visits your page, and ends when the user closes the browser window.
+  </div>
+</popover>
+
+As a developer, storing these usernames and passwords are essential to be able to refer to them later. While there are many ways to store this data - from text files to databases - this article will focus more on how to store passwords properly. For the purpose of this article, we will assume that you are storing the passwords in a database. Modern password storage mechanisms typically involve **salting** and **hashing** the password. This article will describe what these mean and why they are important.
+
+### What's wrong with <trigger for="pop:plaintext">plaintext</trigger>?
+
+<popover id="pop:plaintext" title="_Plaintext_ refers to unencrypted information" placement="top">
+  <div slot="content">
+It is a cryptography term generally referring to text before encyption or after decrypting it. Another term for it is cleartext.
+  </div>
+</popover>
+
+Since usernames are typically public information, it is OK to store them in plaintext. However, passwords are typically secret information and hence, cannot be stored in plaintext.
+
+If passwords are stored in plaintext:
+* Should your database be compromised, attackers can easily use the stored plaintext passwords to login as the user and potentially access other functionality. This grants the attacker more power than just reading values off the database. If the passwords aren't even in plaintext, the attacker might not be able to read the passwords.
```

Comment:
`the attacker might not be able to read the passwords` sounds a bit weak. `should not be able`?